src/FulfillerMultiSelect.jsx

```diff
@@ -62,26 +64,41 @@ class FulfillerMultiSelect extends React.Component {
     refreshComponentData() {
         return Promise.all([
             !this.props.fulfillers
-                ?
-                this.fetchFulfillers(this.props.accessToken)
-                :
-                Promise.resolve(),
+                ? this.fetchFulfillers(this.props.accessToken)
+                : Promise.resolve(),
             this.getRecentFulfillerIds()
         ])
-            .then(() => {
-                if (this.state.recentFulfillerIds.length) {
-                    this.setState(
-                        {
-                            selectedFulfillerIds: this.props.autoSelectMostRecent ?
-                                this.state.recentFulfillerIds :
-                                null
-                        },
-                        () => {
-                            if (this.props.onChange && this.props.autoSelectMostRecent) {
-                                this.props.onChange(this.state.recentFulfillerIds.map(f => this.fulfillerMap[f]));
-                            }
-                        });
+            .then(results => {
+                if (!results[1]) {
+                    return;
                 }
+
+                const { recentFulfillerIds, recentFulfillerIdsMulti } = results[1];
+                let fulfillerIdsToSelect = [];
+                if (this.props.selectedFulfillerIds) {
+                    fulfillerIdsToSelect = this.props.selectedFulfillerIds
+                } else if (recentFulfillerIds.length > 0 && this.props.autoSelectMostRecent) {
+                    fulfillerIdsToSelect = this.props.multi
+                        ? recentFulfillerIdsMulti
```

Comment:
Another change I made since the review is to make `[]` a valid selection for multiselect (i.e. "select all possible fulfillers"). I talked with my team and apparently there are users who take advantage of this and it would be rather annoying to make them always clear the default selection.